marconi-chain-index/src/Marconi/ChainIndex/Indexers.hs

```diff
@@ -617,9 +627,11 @@ runIndexers
   -> IO ()
 runIndexers socketPath networkId cliChainPoint indexingDepth (ShouldFailIfResync shouldFailIfResync) traceName indexerList = do
   securityParam <- Utils.toException $ Utils.querySecurityParam @Void networkId socketPath
+  currentNodeBlockNo <- Utils.toException $ Utils.queryCurrentNodeBlockNo @Void networkId socketPath
```

Comment:
as the `currentNodeBlockNo` is specific to one worker, can't we resolve it in `main` pass it to the `epochStateWorker` (using partial application), and keep the other indexer clean?

Reply:
I initially did that, but it thought that it "corrupted" the `main` with too much indexer specific logic. Then, I thought that this `BlockNo` _could_ be beneficial to other indexers. 

I don't like either of these solutions though, but the idea of keeping `main` clearer "felt" better. I got no objective argument here.

Given my statement, you still think it makes more sense to make the currentNodeBlockNo argument specific to the `EpochState` worker?

Reply:
we could also consider passing the `networkId` and the `socketPath` to the `epochState` and make it monadic. Is it better?
If you still think that the current solution is the least bad one, I'm fine with it, we'll see if we also have to deal with it (and if we can find a better alternative) with the new indexers.

Reply:
it will return an `IO Worker` instead of a `Worker`

Reply:
Like we discussed, we will revisit this on the new API version.